sql/moz-fx-data-shared-prod/firefox_ios_derived/firefox_ios_clients_v1/query.sql

```diff
@@ -0,0 +1,265 @@
+        NULLIF(metrics.string.adjust_ad_group, "") AS adjust_ad_group,
+        NULLIF(metrics.string.adjust_campaign, "") AS adjust_campaign,
+        NULLIF(metrics.string.adjust_creative, "") AS adjust_creative,
+        NULLIF(metrics.string.adjust_network, "") AS adjust_network,
+      FROM
+        firefox_ios.first_session
+      WHERE
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+        AND ping_info.seq = 0 -- Pings are sent in sequence, this guarantees that the first one is returned.
+    )
+  GROUP BY
+    client_id
+),
+-- Find earliest data per client from the metrics ping.
+metrics_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    DATETIME(
+      MIN(submission_timestamp)
+    ) AS min_submission_datetime,  -- do we need this? Looks like later we might want to see timestamp corresponding to the row that contains the campaign info
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    (
+      SELECT
+        client_info.client_id AS client_id,
+        sample_id,
+        submission_timestamp,
+        NULLIF(fxa_metrics.metrics.string.adjust_ad_group, "") AS adjust_ad_group,
+        NULLIF(fxa_metrics.metrics.string.adjust_campaign, "") AS adjust_campaign,
+        NULLIF(fxa_metrics.metrics.string.adjust_creative, "") AS adjust_creative,
+        NULLIF(fxa_metrics.metrics.string.adjust_network, "") AS adjust_network,
+        -- NULLIF(metrics.string.metrics_install_source, "") AS install_source,
+      FROM
+        firefox_ios.metrics AS fxa_metrics
+      WHERE
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+    )
+  GROUP BY
+    client_id
+),
+_current AS (
+  SELECT
+    * EXCEPT (min_submission_datetime, first_run_date, sample_id, adjust_info),
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    COALESCE(first_seen.sample_id, first_session.sample_id, metrics.sample_id) AS sample_id,
+    activated AS activated,
+    STRUCT(
+      IF(first_session.client_id IS NULL, FALSE, TRUE) AS reported_first_session_ping,
+      IF(metrics.client_id IS NULL, FALSE, TRUE) AS reported_metrics_ping,
+      DATE(first_session.min_submission_datetime) AS min_first_session_ping_submission_date,
+      DATE(first_session.first_run_date) AS min_first_session_ping_run_date,
+      DATE(
+        metrics.min_submission_datetime
+      ) AS min_metrics_ping_submission_date, -- TODO: are these date fields useful?
```

Comment:
I haven't used the date fields. Lucia could have been using them to help determine cases when a first_session ping came after the metrics ping.

Reply:
+1, I don't think it's part of the use case for this table, it's meant more as a general understanding of patterns of late arriving data when looking at historical data. The way the table is created incrementally we won't have the edge cases once it starts the daily build.

Reply:
ok, for now I commented these out. We can always re-add those fields.